File: java/vote/src/com/anders/vote/dao/GenericDao.java
package com.anders.vote.dao;

import java.io.Serializable;

import com.anders.vote.mapper.GenericMapper;

public interface GenericDao<PK extends Serializable, T> extends GenericMapper<PK, T> {
	GenericMapper<PK, T> getMapper();

	/**
	 * 保存或更新BO对象
	 * 
	 * @param entity
	 *            BO对象
	 */
	void saveOrUpdate(final T entity);

	/**
	 * 根据主键ID禁用BO对象
	 * 
	 * @param id
	 *            主键ID
	 */
	void disabledById(final PK id);

	/**
	 * 根据主键ID删除BO对象
	 * 
	 * @param id
	 *            主键ID
	 */
	void deleteById(final PK id);

	/**
	 * 根据主键ID判断BO对象是否存在
	 * 
	 * @param id
	 *            主键ID
	 * @return BO对象是否存在（true：存在；false：不存在）
	 */
	boolean isExistById(final PK id);
}

File: fmall/fmall/Dao/GenericDao.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Spring.Data.NHibernate.Generic.Support;

namespace fmall.Dao
{
    public class GenericDao<T, PK> : HibernateDaoSupport, IGenericDao<T, PK>
    {


        public void save(T entity)
        {
            HibernateTemplate.Save(entity);
        }

        public void update(T entity)
        {
            HibernateTemplate.Update(entity);
        }

        public void delete(T entity)
        {
            HibernateTemplate.Delete(entity);
        }

        public void saveOrUpdate(T entity)
        {
            HibernateTemplate.SaveOrUpdate(entity);
        }

        public T getById(PK id)
        {
            return HibernateTemplate.Get<T>(id);
        }

        public void deleteById(PK id)
        {
          T entity = getById(id);
           HibernateTemplate.Delete(entity);
        }

        public void disabledById(PK id)
        {
            //T entity = getById(id);
            //entity
        }

        public IList<T> getByIds(ICollection<PK> ids)
        {
            throw new NotImplementedException();
        }

        public IList<T> findBy(string propertyName, object value)
        {
            throw new NotImplementedException();
        }

        public System.Collections.IList findBy(string propertyName, object value, string selectPropertyName)
        {
            throw new NotImplementedException();
        }

        public T findUniqueBy(string propertyName, object value)
        {
            throw new NotImplementedException();
        }
    }
}
